src/ol/layer/layer.js

```diff
@@ -172,6 +172,7 @@ ol.layer.Layer.prototype.setMap = function(map) {
         map, ol.render.EventType.PRECOMPOSE, function(evt) {
           var layerState = this.getLayerState();
           layerState.managed = false;
+          layerState.zIndex = 1000;
```

Comment:
There are two arbitrary values, `0` and `1000`, which delimitate three main ranges for the user:
- below 0 for layers to place under all others;
- above 1000 for layers to place on top of all others, including unmanaged layers;
- between 0 and 1000 for placing between them.
The proposition of @elemoine to allow setting Z-index on user unmanaged layers fits well with this vision. It provides the maximum flexibility to the user.